Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -1261,36 +1262,32 @@ static inline bool shouldAddContentKeyRecipients()
 void SourceBufferPrivateAVFObjC::configureVideoRenderer(VideoMediaSampleRenderer& videoRenderer)
 {
     videoRenderer.setResourceOwner(m_resourceOwner);
-    RetainPtr displayLayer = dynamic_objc_cast<AVSampleBufferDisplayLayer>(videoRenderer.renderer());
 #if ENABLE(ENCRYPTED_MEDIA) && HAVE(AVCONTENTKEYSESSION)
-    if (m_cdmInstance && shouldAddContentKeyRecipients() && displayLayer)
-        [m_cdmInstance->contentKeySession() addContentKeyRecipient:displayLayer.get()];
+    if (m_cdmInstance && shouldAddContentKeyRecipients() && videoRenderer.as<AVSampleBufferDisplayLayer>())
+        [m_cdmInstance->contentKeySession() addContentKeyRecipient:videoRenderer.as<AVSampleBufferDisplayLayer>()];
 #endif
     videoRenderer.requestMediaDataWhenReady([weakThis = ThreadSafeWeakPtr { *this }] {
         if (RefPtr protectedThis = weakThis.get(); protectedThis && protectedThis->m_enabledVideoTrackID)
             protectedThis->didBecomeReadyForMoreSamples(*protectedThis->m_enabledVideoTrackID);
     });
-    if (displayLayer)
-        m_listener->beginObservingVideoRenderer(displayLayer.get());
+    m_listener->beginObservingVideoRenderer(videoRenderer.renderer());
 }
 
 void SourceBufferPrivateAVFObjC::invalidateVideoRenderer(VideoMediaSampleRenderer& videoRenderer)
 {
     videoRenderer.flush();
     videoRenderer.stopRequestingMediaData();
-    if (RetainPtr renderer = videoRenderer.renderer())
-        m_listener->stopObservingVideoRenderer(renderer.get());
+    m_listener->stopObservingVideoRenderer(videoRenderer.renderer());
```

Comment:
Ditto.

Reply:
`Ref<WebAVSampleBufferListener> m_listener;`

Reply:
should make it const while at it.